tests/integration/relation_tests/test_charm_relations.py

```diff
@@ -91,7 +109,132 @@ async def test_user_with_extra_roles(ops_test: OpsTest):
     client.admin.command(
         "createUser", "newTestUser", pwd="Test123", roles=[{"role": "readWrite", "db": database}]
     )
-
     client["new_database"]
-
     client.close()
+
+
+async def test_two_applications_doesnt_share_the_same_relation_data(
+    ops_test: OpsTest, application_charm
+):
+    """Test that two different application connect to the database with different credentials."""
+    # Set some variables to use in this test.
+    another_application_app_name = "another-application"
+    all_app_names = [another_application_app_name]
+    all_app_names.extend(APP_NAMES)
+
+    # Deploy another application.
+    await ops_test.model.deploy(
+        application_charm,
+        application_name=another_application_app_name,
+    )
+    await ops_test.model.wait_for_idle(apps=all_app_names, status="active")
+
+    # Relate the new application with the database
+    # and wait for them exchanging some connection data.
+    await ops_test.model.add_relation(
```

Comment:
To ensure your tests are independent of each other, you could remove your relations after each test; this way, they aren't a confounding variable in the following tests, which would be hard to debug.

Reply:
PMing you!!